src/Compilers/Core/Portable/Collections/DictionaryExtensions.cs

```diff
@@ -48,5 +52,29 @@ internal static class DictionaryExtensions
             return true;
         }
 #endif
+
+        public static void AddPooled<K, V>(this IDictionary<K, ArrayBuilder<V>> dictionary, K key, V value)
```

Comment:
>AddPooled [](http://example.com/codeflow?start=27&length=9)

These two methods seem very specialized. Consider defining them in the one caller instead.

Reply:
They are not that specialized. Similar pattern is used in multiple places. I'll follow up with updating the other use sites, didn't want to clutter this PR with such changes.